src/target/riscv/riscv.c

```diff
@@ -5029,6 +5040,72 @@ int riscv_dmi_write_u64_bits(struct target *target)
 	return r->dmi_write_u64_bits(target);
 }
 
+static int check_if_trigger_exists(struct target *target, unsigned int t)
+{
+	/* If we can't write tselect, then this hart does not support triggers. */
+	if (riscv_set_register(target, GDB_REGNO_TSELECT, t) != ERROR_OK)
+		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
+	riscv_reg_t tselect_rb;
+	if (riscv_get_register(target, &tselect_rb, GDB_REGNO_TSELECT) != ERROR_OK)
+		return ERROR_FAIL;
+	/* Mask off the top bit, which is used as tdrmode in old
+	 * implementations. */
+	tselect_rb &= ~(1ULL << (riscv_xlen(target) - 1));
+	if (tselect_rb != t)
+		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
+	return ERROR_OK;
+}
+
+static int get_trigger_types(struct target *target, unsigned int *trigger_tinfo,
+		riscv_reg_t tdata1)
+{
+	assert(trigger_tinfo);
+	riscv_reg_t tinfo;
+	if (riscv_get_register(target, &tinfo, GDB_REGNO_TINFO) == ERROR_OK) {
+		if (get_field(tinfo, CSR_TINFO_INFO) == 1)
+			/* trigger doesn’t exist */
+			return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
```

Comment:
This check has been made less defensive instead of more defensive.

The original code checked for  `tinfo==1` and `tinfo==0`.

Please, if you do not wish to extend the condition to `tinfo & 0x1`, leave at least the current state, unless there is very good reason to make the check weaker.

Reply:
Addressed.